internal/daemon/pipeline/pipeline.go

```diff
@@ -27,7 +27,7 @@ type Pipelines []Pipeline
 type Pipeline struct {
 	ID      string `yaml:"pipeline.id"`
 	Config  string `yaml:"path.config"`
-	Ordered string `yaml:"pipeline.ordered"`
+	Ordered string `yaml:"-"`
```

Comment:
Why are we keeping this field around if we're not including it in the generated config file?

Reply:
The field `Ordered` in `pipelines.yml` is only available in Logstash >= 7.7.0. I am not completely sure, but it is my assumption, that it would be beneficial to set `Ordered` to `true`, if Logstash >= 7.7.0 is used, even though I did not observe any issues without it so far. Therefore we need to generate a different version of the `pipelines.yml` depending on the Logstash version in use (unfortunately, the validation of the `pipelines.yml` in Logstash fails in older versions, if there are unknown keys).

So the bottom line is: You are correct, `Ordered` is currently not used and could be removed, but I would like to add it for Logstash >= 7.7.0. The reason I have not yet implemented this is, that I am unsure about the best way to do it.

What do you propose? Remove it? Add comment? Do you have an idea how to implement the different generation of the yaml depending on the Logstash version?

Reply:
How about adding the `omitempty` flag and only setting Ordered to true when running Logstash >= 7.7?

Reply:
@magnusbaeck I updated the PR with your proposed solution (and added a small feature to prevent LFV in daemon mode from cleaning the temporary files for debug purposes).